kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/Eq.kt

```diff
@@ -19,5 +19,17 @@ fun <T> eq(actual: T, expected: T): Throwable? = when {
    actual is Regex && expected is Regex -> RegexEq.equals(actual, expected)
    actual is String && expected is String -> StringEq.equals(actual, expected)
    actual is Number && expected is Number -> NumberEq.equals(actual, expected)
+   // HELP! How to get project config here? I was thinking...
+   // Project.showDetailedDataClassDiff() && actual.isDataClass() ...
```

Comment:
ProjectConfig is only part of kotest-core (the test runner) so we can't use it in the assertion library.
I would just have a top level var like I do for `var shouldRemoveKotestElementsFromStacktrace: Boolean = sysprop("kotest.failures.stacktrace.clean", "true") == "true"`
It's unlikely that people are overriding equals, so I would default to true.

Reply:
It might be worth starting an AssertionsConfig object, and putting the var in there.